java/bulk-export/bulk-export-core/src/test/java/sleeper/bulkexport/model/BulkExportLeafPartitionQuerySerDeTest.java

```diff
@@ -0,0 +1,267 @@
+        // Given
+        String tableId = "t-id";
+        tableProperties.set(TABLE_ID, tableId);
+
+        String exportId = "e-id";
+        String subExportId = "se-id";
+        String leafPartitionId = "lp-id";
+        RangeFactory rangeFactory = new RangeFactory(schema);
+        Region region1 = new Region(rangeFactory.createRange(field, 1, true, 10, true));
+        Region partitionRegion = new Region(rangeFactory.createRange(field, 0, 1000));
+        BulkExportLeafPartitionQuery bulkExportLeafPartitionQuery = BulkExportLeafPartitionQuery.builder()
+                .tableId(tableId)
+                .exportId(exportId)
+                .subExportId(subExportId)
+                .regions(List.of(region1))
+                .leafPartitionId(leafPartitionId)
+                .partitionRegion(partitionRegion)
+                .files(Collections.singletonList("/test/file.parquet"))
+                .build();
+
+        BulkExportLeafPartitionQuerySerDe querySerDe = generateQuerySerDeFromSchema(schema);
+
+        // When
+        String json = querySerDe.toJson(bulkExportLeafPartitionQuery, true);
+        BulkExportLeafPartitionQuery deserialisedQuery = querySerDe.fromJson(json);
+
+        // Then
+        assertThat(bulkExportLeafPartitionQuery).isEqualTo(deserialisedQuery);
+        Approvals.verify(json, new Options().forFile().withExtension(".json"));
+    }
+
+    @Test
+    public void shouldThrowExceptionWithNullTableId() {
+        // Given
+        String tableId = "t-id";
+        tableProperties.set(TABLE_ID, tableId);
+
+        String exportId = "e-id";
+        String subExportId = "se-id";
+        String leafPartitionId = "lp-id";
+        BulkExportLeafPartitionQuerySerDe querySerDe = generateQuerySerDeFromTableProperties(schema);
+
+        // When / Then
+        assertThatThrownBy(() -> querySerDe.toJson(BulkExportLeafPartitionQuery.builder()
+                .exportId(exportId)
+                .subExportId(subExportId)
+                .leafPartitionId(leafPartitionId)
+                .files(Collections.singletonList("/test/file.parquet"))
+                .build()))
+                .isInstanceOf(
+                        BulkExportQueryValidationException.class)
+                .hasMessage("Query validation failed for export \"" + exportId + "\": " +
+                        "tableId field must be provided");
+    }
+
+    @Test
+    public void shouldThrowExceptionNoExportId() throws IOException {
+        // Given
+        String tableId = "t-id";
+        tableProperties.set(TABLE_ID, tableId);
+        BulkExportLeafPartitionQuerySerDe querySerDe = generateQuerySerDeFromTableProperties(schema);
+
+        String json = readFileToString(
+                "src/test/java/sleeper/bulkexport/model/BulkExportLeafPartitionQuerySerDeTest.shouldThrowExceptionNoExportId.approved.json");
+
+        // When & Then
+        Approvals.verify(json, new Options().forFile().withExtension(".json"));
```

Comment:
It looks like this is just asserting that the file that was read is the file that was read.

I'm also not sure we can rely on the working directory always being the same when these tests are run.

I think Approvals.verify seems right for the first two tests, but it doesn't really apply when we're testing the validation rather than the deserialisation. For the rest of these, can we remove this assertion and put the file in the test resources, like with ClientTestUtils.example?

Reply:
Json moved

Reply:
It's still using the working directory rather than loading it from resources. I'll add a separate comment.